### PR TITLE
bumper: Move github actions to matrix

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
With matrix strategy we need just one github action that runs on all the
release branches and main.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/component-bumper.yaml [hunk 1]
```diff
@@ -5,26 +5,30 @@ on:
...
-        ref: ${{ env.BASE_BRANCH }}
+        ref: ${{ matrix.branch }}
 
     - name: Pull latest of latest branch
-      run:  git pull --ff-only --rebase origin ${{ env.BASE_BRANCH }}
+      run:  git pull --ff-only --rebase origin ${{ matrix.branch }}
 
     - name: Run bumper script
-      run: make ARGS="-config-path=components.yaml -token=${{ secrets.GITHUB_TOKEN }} -base-branch=${{ env.BASE_BRANCH }}" auto-bumper
+      run: make ARGS="-config-path=components.yaml -token=${{ secrets.GITHUB_TOKEN }} -base-branch=${{ matrix.branch }}" auto-bumper
```
Comment (rhrazdil) [inline .github/workflows/component-bumper.yaml:34]: Don't we want to use the KUBEVIRT_BOT_TOKEN?

File: .github/workflows/component-bumper.yaml [hunk 2]
```diff
@@ -5,26 +5,30 @@ on:
...
     if: (github.repository == 'kubevirt/cluster-network-addons-operator')
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        branche:
+          - main
+          - release-0.42
+          - release-0.44
+          - release-0.53
+          - release-0.58
```
Comment (rhrazdil) [inline .github/workflows/component-bumper.yaml:19]: aren't we missing release-0.65 here?

File: .github/workflows/component-bumper.yaml (hunk 1 above)
Comment (qinqon) [reply .github/workflows/component-bumper.yaml:34]: Need rebase

File: .github/workflows/component-bumper.yaml (hunk 2 above)
Comment (qinqon) [reply .github/workflows/component-bumper.yaml:19]: Need rebase

**Comment by qinqon** [comment]
/override pull-e2e-cluster-network-addons-operator-kubemacpool-functests
/override pull-e2e-cluster-network-addons-operator-nmstate-functests
/override pull-e2e-cluster-network-addons-operator-workflow-k8s

**Pull request merged**